Firestore/core/test/firebase/firestore/model/precondition_test.cc

```diff
@@ -33,10 +33,11 @@ TEST(Precondition, None) {
   EXPECT_TRUE(none.IsNone());
   EXPECT_EQ(SnapshotVersion::None(), none.update_time());
 
-  const NoDocument deleted_doc = testutil::DeletedDoc("foo/doc", 1234567);
-  const Document doc = testutil::Doc("bar/doc", 7654321);
-  EXPECT_TRUE(none.IsValidFor(&deleted_doc));
-  EXPECT_TRUE(none.IsValidFor(&doc));
+  const std::shared_ptr<NoDocument> deleted_doc =
```

Comment:
Optional: there was previously an agreement to avoid `const` values, so consider omitting those `const`s. Entirely optional, because they weren't added in this PR.

Reply:
Done throughout this file. I've not attempted to apply this more broadly though.